lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocations.java

```diff
@@ -62,4 +62,94 @@ public void testInvalidLocationInAppendPath(String location, String exceptionMes
                 .isInstanceOf(IllegalArgumentException.class)
                 .hasMessageMatching(exceptionMessageRegexp);
     }
+
+    private static Stream<Arguments> testExtractVariousInfoData()
+    {
+        return Stream.of(
+                Arguments.of("", "", null, ""),
+                Arguments.of("test_file", "test_file", null, "test_file"),
+                Arguments.of("test>file", "test>file", null, "test%3Efile"),
+                Arguments.of("test_dir/", "", null, "test_dir/"),
+                Arguments.of("/test_file.txt", "test_file.txt", null, "/test_file.txt"),
+                Arguments.of("test_dir/test_file.txt", "test_file.txt", null, "test_dir/test_file.txt"),
+                Arguments.of("test_dir /test_file.txt", "test_file.txt", null, "test_dir%20/test_file.txt"),
+                Arguments.of("test_dir  /test_file.txt", "test_file.txt", null, "test_dir%20%20/test_file.txt"),
+                Arguments.of("test_<dir  /test_file.txt", "test_file.txt", null, "test_%3Cdir%20%20/test_file.txt"),
+                Arguments.of("test_dir/test_dir2/", "", null, "test_dir/test_dir2/"),
+                Arguments.of("s3://test_dir/test_file.txt", "test_file.txt", "s3", "s3://test_dir/test_file.txt"),
+                Arguments.of("s3://test_dir/test_dir2/test_file.txt", "test_file.txt", "s3", "s3://test_dir/test_dir2/test_file.txt"),
+                Arguments.of("s3://dir_with_space /test_file.txt", "test_file.txt", "s3", "s3://dir_with_space%20/test_file.txt"),
+                Arguments.of("file://test_dir/test_file", "test_file", "file", "file://test_dir/test_file"),
+                Arguments.of("file:/test_dir/test_file", "test_file", "file", "file:/test_dir/test_file"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("testExtractVariousInfoData")
+    public void testExtractVariousInfo(String location, String fileName, String scheme, String uri)
+    {
+        assertThat(Locations.getFileName(location)).isEqualTo(fileName);
+        assertThat(Locations.getScheme(location)).isEqualTo(scheme);
+        assertThat(Locations.getUri(location).toString()).isEqualTo(uri);
```

Comment:
Why do you test against the string representation of the `URI` and not the `URI` itself?

Reply:
Good question. I will change that.